apps/src/lib/ui/lti/registration/LtiDynamicRegistrationPage.tsx

```diff
@@ -0,0 +1,94 @@
+import Typography from '@cdo/apps/componentLibrary/typography';
+import React, {useState} from 'react';
+import {Button} from '@cdo/apps/componentLibrary/button';
+import i18n from '@cdo/locale';
+import styles from './styles.module.scss';
+import $ from 'jquery';
+
+interface LtiDynamicRegistrationProps {
+  logoUrl: string;
+  registrationID: string;
+  onSubmit: () => void;
+}
+
+export const LtiDynamicRegistrationPage = ({
+  logoUrl,
+  registrationID,
+  onSubmit,
+}: LtiDynamicRegistrationProps) => {
+  const [submitDisable, setSubmitDisable] = useState<boolean>(false);
+  const [email, setEmail] = useState('');
+  const [hasError, setHasError] = useState<boolean>(false);
+  const [errorMsg, setErrorMsg] = useState('');
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    setSubmitDisable(true);
+
+    $.ajax({
+      type: 'POST',
+      url: '/lti/v1/dynamic_registration',
+      data: {
+        email: email,
+        registration_id: registrationID,
+      },
+      dataType: 'json',
+      success: () => {
+        if (typeof onSubmit === 'function') {
+          onSubmit();
```

Comment:
Curious why you moved this up one level, it's still good but just wondering

Reply:
I was following another example in our code, thinking it may be best practice to have a method like this passed in via the component props, as it seems to make it easier to test/mock. Definitely not a React pro, and heavily relied on other examples in our code.

Reply:
Roger that (and agreed), I moved it back.